Makefile

```diff
@@ -28,6 +28,13 @@ update:
 	go get -t -v -d -u ./...
 	@echo "Dependencies updated"
 
+.PHONY: update
+update-schema:
+	go build -v -tags 'libsqlite3 devel' ./lxd/schema.go
```

Comment:
What's the devel tags for?

And "libsqlite3" isn't guaranteed to work everywhere so you should use the $TAGS variable instead which already has the detection logic.

Reply:
Why not? Isn't the sqlite fork of dqlite backward compatible with regular libsqlite3?

We obviously can't have the sqlite driver use one version of libsqlite3 and dqlite use another, but there shouldn't be any problem with having both use the dqlite build of libsqlite3.

Reply:
@stgraber yes the fork is fully compatible, as it only adds a few new APIs. The problem is the github.com/mattn/go-sqlite3 package, which, if you don't set the libsqlite3 tag, will unconditionally build and link against its [local copy](https://github.com/mattn/go-sqlite3/blob/master/sqlite3-binding.c) of sqlite. AFAIU there's no way to avoid that, or at least I've found none that does not require changing the go-sqlite3 package.

Reply:
Ok, so yeah, I guess we'll have to force the use of the libsqlite3 tag then. Also hoping that we'll make progress with upstreaming the hook and go-sqlite3 API change so that we can get rid of our forks of those two.

Reply:
I made a post to the sqlite-users mailing list, presenting dqlite and pointing to the patch. I said that if there was interest in merging the patch upstream, I'd go through whatever process is required. There was a guy expressing interest in dqlite itself and asking some questions, but no upstream author has stepped up. You can read the thread [here](https://pastebin.com/3byUqbe2).

My plan would be to wait for integration with LXD to be finished, then exercise the whole thing end-to-end both in real world and with automated stress tests, fix any bug that comes up, and finally go back to sqlite authors saying that we got something solid. Hopefully there will be some more trust/interest at that point.